### PR TITLE
civl 1.7-3157 (new formula)

### PR DESCRIPTION
**Pull request by AlshammariA**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?
### Description

This is a new formula for CIVL, this is the first version of the software in Homebrew

### CODE REVIEW

**Comment by AlshammariA** [comment]
Wish to give me your feedback ..

File: Formula/civl.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+
+class Civl < Formula
```
Comment (UniqMartin) [inline Formula/civl.rb]: Please remove the empty line at the top of the file.

File: Formula/civl.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  desc "The Concurrency Intermediate Verification Language"
+  homepage "http://vsl.cis.udel.edu/civl/"
+  url "http://vsl.cis.udel.edu/lib/sw/civl/1.7/r3157/release/CIVL-1.7_3157.tgz"
+  sha256 "33fa5b4877fbf494b0a4b426c15d6a165fa3db48a9f97cbf391818d965d2dfb9"
+  depends_on :java => "1.8+"
```
Comment (UniqMartin) [inline Formula/civl.rb]: Please leave an empty line between the `sha256` line and the first `depends_on`.

File: Formula/civl.rb [hunk 3]
```diff
@@ -0,0 +1,22 @@
+
+class Civl < Formula
+  desc "The Concurrency Intermediate Verification Language"
+  homepage "http://vsl.cis.udel.edu/civl/"
+  url "http://vsl.cis.udel.edu/lib/sw/civl/1.7/r3157/release/CIVL-1.7_3157.tgz"
```
Comment (UniqMartin) [inline Formula/civl.rb]: Homebrew has trouble auto-detecting the version number from the URL. Please add a `version "1.7_3157"` line after this one.

File: Formula/civl.rb [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+  def install
+    libexec.install "lib/civl-1.7_3157.jar"
+    bin.write_jar_script libexec/"civl-1.7_3157.jar", "civl"
+    share.install "doc"
+    share.install "examples"
+    share.install "licenses"
+    share.install "emacs"
+  end
+  test do
```
Comment (UniqMartin) [inline Formula/civl.rb]: Please leave an empty line between `end` and `test do`.

File: Formula/civl.rb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    system "#{bin}/civl"
```
Comment (UniqMartin) [inline Formula/civl.rb]: Can the test be modified to do something more substantial than this? [This section of the Formula Cookbook](https://github.com/Homebrew/brew/blob/master/share/doc/homebrew/Formula-Cookbook.md#add-a-test-to-the-formula) should get you started, and I'm happy to answer any questions you might have.

File: Formula/civl.rb [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  depends_on "z3"
+  depends_on "cvc4"
+
+  def install
+    libexec.install "lib/civl-1.7_3157.jar"
+    bin.write_jar_script libexec/"civl-1.7_3157.jar", "civl"
+    share.install "doc"
+    share.install "examples"
+    share.install "licenses"
+    share.install "emacs"
```
Comment (UniqMartin) [inline Formula/civl.rb]: `install` accepts multiple arguments. You can replace the above four lines with this:

```ruby
share.install "doc", "examples", "licenses", "emacs"
```

File: Formula/civl.rb (hunk 5 above)
Comment (AlshammariA) [inline Formula/civl.rb]: At this point, I just want to make it like this as the first formula. Then I will enhance it later. Thanks for your commits

File: Formula/civl.rb (hunk 2 above)
Comment (UniqMartin) [inline Formula/civl.rb]: Also, this formula doesn't build anything, which means we don't need to create binary packages (bottles) for it. Please add a `bottle :unneeded` line between the `sha256` and the `depends_on` line. (It should be surrounded by empty lines.)

**Comment by AlshammariA** [comment]
If I update the formula based on your comments. I can't pass the 
brew audit --strict --online civl

Here is the error list:

Stable: version 1.7_3157 should not end with an underline and a number
  * 7: Trailing whitespace was found
  * 9: Trailing whitespace was found
  * 19: Trailing whitespace was found
  * C: 7: col 1: Trailing whitespace detected.
  * C: 9: col 1: Trailing whitespace detected.
  * C: 19: col 1: Trailing whitespace detected.

**Comment by UniqMartin** [comment]
> Stable: version 1.7_3157 should not end with an underline and a number

You are right. Sorry, my fault! We will need to make the `version` line 1.7-r3157 or 1.7-3157 for it to work. I think the former is preferable, as the “r3157” also appears in the download URL.

> 7: Trailing whitespace was found

This and the following errors you should be able to address yourself, I think. Many editors also offer an option to automatically remove trailing whitespace on save or on demand, to help you with that.

**Comment by AlshammariA** [comment]
Thank you for your support. As soon As I fix everything, I will let you know.

File: Formula/civl.rb (hunk 5 above)
Comment (UniqMartin) [inline Formula/civl.rb]: Cannot one of the examples that also get installed be used to run a simple test? I'm of course not familiar with the software, so I don't know what is possible. But I assume you're a user, so could have some ideas. Thanks for maybe reconsidering and giving it a try!

Comment (AlshammariA) [inline Formula/civl.rb]: You are right. I think I will use an example to test. Let me try. Should I close the request? or I keep it open?

Comment (UniqMartin) [inline Formula/civl.rb]: Thank you! Please leave the pull request open. It's fine if it takes you some time to make all the adjustments and to add the test. Simply push your changes once you're done and ideally leave a comment and then I or some other maintainer will notice and get back to you.

Comment (AlshammariA) [inline Formula/civl.rb]: Awesome. Thanks for your support.

Comment (apjanke) [inline Formula/civl.rb]: Yep! In fact, definitely leave a comment when you think this is ready for another review. We use the "needs response" label to indicate that we're waiting for a response or update from the submitter. When you leave a comment on the PR, that label will go away and we'll notice that this PR needs attention and get back to you.

Thanks for your contribution and for working with us on this!

**Comment by UniqMartin** [comment]
@AlshammariA At this point I'm waiting for you to improve the test. Or did we misunderstand each other? I saw you addressed all other feedback and other than the test the formula is looking great.

**Comment by AlshammariA** [comment]
@UniqMartin I got your idea in the first time. I think I haven't explained very well. I need time to write the test and improve it because I will meet an expert in this language who will explain to me more. At this point, if it is not REQUIRED to improve the test, I would like to improve the test when I update the the formula, probably next month since it will be a new version of this tool. During this time, I can learn how to improve the test by adding an example. If we cannot publish the first version without improving the test, please let me know, so we can close this request and do another one as soon as I get the test improved.

**Comment by mikemcquaid** [comment]
> If we cannot publish the first version without improving the test, please let me know, so we can close this request and do another one as soon as I get the test improved.

Closing as requested, thanks.

**Pull request closed without merging**